### PR TITLE
library/serato/seratofeature: Fix relative paths on OSX (and on Linux)

### PR DESCRIPTION
**Pull request by Holzhaus**
Fixes the path issue on OSX that was reported here: https://github.com/mixxxdj/mixxx/pull/2480#issuecomment-612677556

### CODE REVIEW

**Comment by bengl3rt** [comment]
Unfortunately still not able to load tracks from the main Serato DB on the boot drive. Loading one from the external drive still works. 

Attaching a debug log where I first tried to load a track from the main DB, then from the external.
[serato.txt](https://github.com/mixxxdj/mixxx/files/4467390/serato.txt)

**Comment by Holzhaus** [comment]
Thanks for checking. You can right click the column headers in the serato library view and un-hide the location column to see the file path detected by mixxx. It's incorrect I assume?

**Comment by Holzhaus** [comment]
Btw, if you're not using CMake please try a clean build to make sure it builds the correct version.

**Comment by bengl3rt** [comment]
Yep, all clean all the time now - though I'm not even sure I trust that (it's scons -c, right?) because the following rebuilds complete suspiciously quickly.

Unhiding the location column makes all the tracks listed in the Serato library disappear, along with some scary looking errors coming out in the console. Debug log attached.
[serato-2.txt](https://github.com/mixxxdj/mixxx/files/4467447/serato-2.txt)

**Comment by Holzhaus** [comment]
> Unhiding the location column makes all the tracks listed in the Serato library disappear, along with some scary looking errors coming out in the console. Debug log attached.
> [serato-2.txt](https://github.com/mixxxdj/mixxx/files/4467447/serato-2.txt)

Ouch. This query looks like it comes from BaseTrackCache. @uklotzde Any ideas what causes them?

**Comment by Holzhaus** [comment]
> > Unhiding the location column makes all the tracks listed in the Serato library disappear, along with some scary looking errors coming out in the console. Debug log attached.
> > [serato-2.txt](https://github.com/mixxxdj/mixxx/files/4467447/serato-2.txt)
> 
> Ouch. This query looks like it comes from BaseTrackCache. @uklotzde Any ideas what causes them?

Ping @uklotzde

**Comment by Holzhaus** [comment]
@bengl3rt Yikes, I used the wrong ifdef. Despite OSX being a unix system, we actually only set the `__UNIX__` compile definition for Linux and BSD only. No wonder it doesn't work for you.

**Comment by Holzhaus** [comment]
@bengl3rt please retest

**Comment by bengl3rt** [comment]
> @bengl3rt please retest

Done and working as expected. Location column shows the correct path for files on internal and external drives, and they play and load as expected.

I think a previous version of this patch, or possibly all the hopping around between different branches I've been doing, severely mangled my database so I did blow out ~/Library/Application Support/Mixxx and let it rescan everything. Upon doing that the scary SQL database error went away.

**Comment by Be-ing** [comment]
Yay! Thanks for testing @bengl3rt!

File: src/library/serato/seratofeature.cpp [hunk 1]
```diff
@@ -450,10 +450,25 @@ QString parseDatabase(mixxx::DbConnectionPoolPtr dbConnectionPool, TreeItem* dat
     databaseRootDir.cdUp();
```
Comment (Be-ing) [inline src/library/serato/seratofeature.cpp:450]: This line is unnecessary now, right?

Comment (Holzhaus) [reply src/library/serato/seratofeature.cpp:450]: No. `QDir:'root()` is a static method and always returns the system drive letter. I think you can move your home directory onto a different drive on Windows, right?

**Comment by uklotzde** [comment]
> > > Unhiding the location column makes all the tracks listed in the Serato library disappear, along with some scary looking errors coming out in the console. Debug log attached.
> > > [serato-2.txt](https://github.com/mixxxdj/mixxx/files/4467447/serato-2.txt)
> > 
> > 
> > Ouch. This query looks like it comes from BaseTrackCache. @uklotzde Any ideas what causes them?
> 
> Ping @uklotzde

Is this issue still unresolved?

**Comment by Holzhaus** [comment]
> > > > Unhiding the location column makes all the tracks listed in the Serato library disappear, along with some scary looking errors coming out in the console. Debug log attached.
> > > > [serato-2.txt](https://github.com/mixxxdj/mixxx/files/4467447/serato-2.txt)
> > > 
> > > 
> > > Ouch. This query looks like it comes from BaseTrackCache. @uklotzde Any ideas what causes them?
> > 
> > 
> > Ping @uklotzde
> 
> Is this issue still unresolved?

@bengl3rt:
> Upon doing that the scary SQL database error went away.

I think not, although it might indicate a potential for users that update from 2.2. If your still have a 2.2 database lying around, it might be a good idea to check that the upgrade doesn't cause any issues.

**Pull request merged**